lib/rules/avoid-leaking-state-in-ember-objects.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+const ember = require('../utils/ember');
+const utils = require('../utils/utils');
+
+
+//------------------------------------------------------------------------------
+// Ember object rule - Avoid leaking state
+// (Don't use arrays or objects as default props)
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Avoids state leakage',
+      category: 'Ember Object',
+      recommended: true
+    },
+    fixable: null, // or "code" or "whitespace"
+    schema: [{
+      type: 'array',
+      items: { type: 'string' },
+    }],
+  },
+
+  create(context) {
+    const defaultIgnoredProperties = [
+      'classNames',
+      'classNameBindings',
+      'actions',
+      'concatenatedProperties',
+      'mergedProperties',
+      'positionalParams',
+      'attributeBindings',
+      'queryParams',
+      'attrs',
+    ];
+
+    let ignoredProperties = context.options[0] || defaultIgnoredProperties;
+
+    const message = 'Only string, number, symbol, boolean, null, undefined, and function are allowed as default properties';
+
+    const report = function (node) {
+      context.report(node, message);
+    };
+
+    const isAllowed = function(property) {
+      const value = property.value;
+      return ember.isFunctionExpression(value) ||
+        utils.isLiteral(value) ||
+        (
+          utils.isIdentifier(value) &&
+          value.name === "undefined"
```

Comment:
I believe that to satisify some of the new test cases I mentioned that this condition will have to be dropped (and we would then allow identifiers)

Reply:
removing check for value.name === "undefined" for Foo.extend({ derp: importedThing }) case.